apps/shop/permissions.py

```diff
@@ -0,0 +1,23 @@
+from rest_framework import (
+    permissions,
+)
+
+from .models import (
+    ORDER_STATUS_INCOMPLETE,
+)
+
+
+class IsOwnerPermission(permissions.BasePermission):
+    def has_object_permission(self, request, view, obj):
+        try:
+            return obj.user.id == request.user.id
```

Comment:
Still not optimized. Explain what the mistake is in reply and fix it, please.

Reply:
Just as [above](https://github.com/Zomba4okk/DjangoOnlinerLikeShop/pull/33#discussion_r612347723), user is a `ForeignKey` field and thus `obj` should have `user_id` field.
So the solution would be: `obj.user_id == request.user.id`